File: c/singles/selection.c
#include <stdio.h>
/* enum in C*/
enum color {RED, GREEN, BLUE, YELLOW, WHITE, BLACK};

int main() {
    /* some variables */
    int x = -5;
    int y = 0;

    /* y should contain info about sign of x*/
    if (x > 0) {
        y = 1;
    }
    else if (x < 0) {
        y = -1;
    }
    else {
        y = 0;
    }

    /* now use y in a switch case */
    switch (y) {
        case 0: 
            x = 0;
            break;  /* break is important */
        case 1:
            x = 1;
            break;
        default:
            x = -1;
            break;
    }

    /* combined conditions */
    if ((x > 0) || (y < 0)) {
        /* shoud not come here ... */
    }

    /* enum variable */
    enum color current_color = GREEN;

    switch (current_color) {
        case BLACK:
            printf("black\n");
            break;
        case RED:
            printf("red\n");
            break;
        default:
            printf("not black and not red\n");
            break;
    }

    return 0;
}

File: py/singles/selection.py
# use enum in Python.
from enum import Enum
# define enum for color
class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3
    YELLOW = 4
    WHITE = 5
    BLACK = 6
# use enum variable
current_color = Color.RED

# selection using match
match(current_color):
    case Color.BLUE:
        print("blue")
    case _:
        print("other than blue")

# selection using if...
x = -5

# y gives a hint about x sign
if x > 0:   
    y = 1
elif x < 0:
    y = -1
else:
    y = 0

# combined condition
if (x < 0) and not (y < 0):
    print ("cannot be.")
else: 
    print ("everything ok.")

print(y)
